### PR TITLE
build-scripts: Refines Guix build scripts

### PR DESCRIPTION
**Pull request by efimerspan**
This does a bit of polishing on the provided Guix build scripts to get rid of input labels, use Gexps, and other tweaks.

### CODE REVIEW

**Comment by aartaka** [comment]
Looks just fine to me! @Ambrevar, any feedback?

**Comment by Ambrevar** [comment]
Looks good to me, except one thing: are you sure that removing `git` works?  Does Nyxt still report the right version (commit) then?

File: build-scripts/nyxt-quicklisp.scm [hunk 1]
```diff
@@ -68,69 +49,67 @@
...
+                (wrap-program bin
+                  `("GIO_EXTRA_MODULES" prefix
+                    (,(string-append #(this-package-input "glib-networking") "/lib/gio/modules")))
+                  `("GI_TYPELIB_PATH" prefix (,gi-path))
+                  `("LD_LIBRARY_PATH" ":" prefix (,path))
+                  `("XDG_DATA_DIRS" ":" prefix (,xdg-path)))
+                #t))))))
     (native-inputs
-     `(("sbcl" ,sbcl)
-       ;; To generate the right version in Nyxt, unneeded from 2.0 onwards:
```
Comment (efimerspan) [inline build-scripts/nyxt-quicklisp.scm:120]: Yup, `nyxt-version` yields the correct Nyxt version without `git`, as per this comment.

**Comment by Ambrevar** [comment]
Excellent work, thanks a lot!

**Pull request merged**